pkg/oc/admin/policy/modify_roles.go

```diff
@@ -513,7 +537,44 @@ func (o *RoleModificationOptions) AddRole() error {
 		}
 	}
 
-	roleBinding.SetSubjects(addSubjects(o.Users, o.Groups, o.Subjects, roleBinding.Subjects()))
+	// warn if binding to non-existent role
+	if o.PrintErrf != nil {
+		var err error
+		if roleBinding.RoleKind() == "Role" {
+			_, err = o.RbacClient.Roles(o.RoleBindingNamespace).Get(roleBinding.RoleName(), metav1.GetOptions{})
+		} else {
+			_, err = o.RbacClient.ClusterRoles().Get(roleBinding.RoleName(), metav1.GetOptions{})
+		}
+		if err != nil && kapierrors.IsNotFound(err) {
+			o.PrintErrf("Warning: role '%s' not found\n", roleBinding.RoleName())
+		}
+	}
+
+	newSubjects := addSubjects(o.Users, o.Groups, o.Subjects, roleBinding.Subjects())
+	// warn if binding to non-existent subject
+	for _, newSubject := range newSubjects {
+		if o.PrintErrf != nil {
```

Comment:
This will warn on every subject even pre-existing ones we are not adding with this call.
A simple way to avoid warning on subjects we are not touching is to check if newSubject is already in roleBinding.Subjects() and skip checking in that case.